suite-native/module-connect-popup/src/hooks/useConnectPopupNavigation.ts

```diff
@@ -28,22 +31,56 @@ const isConnectPopupUrl = (url: string): boolean => {
     return false;
 };
 
+const isWalletConnectUrl = (url: string): boolean =>
+    url.startsWith('trezorsuitelite://walletconnect');
+
 // TODO: will be necessary to handle if device is not connected/unlocked so we probably want to wait until user unlock device
 // we already have some modals like biometrics or coin enabled which are waiting for device to be connected
 export const useConnectPopupNavigation = () => {
     const featureFlagEnabled = useFeatureFlag(FeatureFlag.IsConnectPopupEnabled);
+    const featureFlagWalletConnectEnabled = useFeatureFlag(FeatureFlag.IsWalletConnectEnabled);
     const navigation = useNavigation<NavigationProp>();
+    const dispatch = useDispatch();
+    const connectPopupCall = useSelector(selectConnectPopupCall);
+    const walletConnectProposal = useSelector(selectPendingProposal);
+    const lastProposalId = useRef<number | null>(null);
 
+    // Handle deeplink
     const url = Linking.useURL();
 
     useEffect(() => {
-        if (!featureFlagEnabled) return;
-        if (!url || !isConnectPopupUrl(url)) return;
-        try {
-            const parsedUrl = Linking.parse(url);
-            navigation.navigate(RootStackRoutes.ConnectPopup, { parsedUrl });
-        } catch (error) {
-            console.warn('Invalid deeplink URL', { error, url });
+        if (!featureFlagEnabled || !url) return;
+
+        if (isConnectPopupUrl(url)) {
+            dispatch(connectPopupDeeplinkThunk({ url }));
+        } else if (featureFlagWalletConnectEnabled && isWalletConnectUrl(url)) {
+            try {
+                const parsedUrl = new URL(url);
+                const wcUri = parsedUrl?.searchParams?.get('uri');
+                if (wcUri) dispatch(walletConnectPairThunk({ uri: wcUri }));
+            } catch {
+                // Malformed url, ignore
```

Comment:
I think it would be nice to track this in Sentry or something

Reply:
There's not much we can do about it from our side